homeassistant/components/zha/core/gateway.py

```diff
@@ -161,6 +161,7 @@ async def async_initialize(self) -> None:
         self.radio_description = RadioType[radio_type].description
 
         app_config = self._config.get(CONF_ZIGPY, {})
+        assert self._hass.config.config_dir
         database = self._config.get(
             CONF_DATABASE,
             os.path.join(self._hass.config.config_dir, DEFAULT_DATABASE_NAME),
```

Comment:
We can use `hass.config.path` instead to join the config dir path. Then we can remove the assert above.